src/Default/default.ts

```diff
@@ -88,7 +165,23 @@ export class DefaultContainer extends WebContainerBase {
             }
         }
 
-        return this.wrapWindow(this.globalWindow.open(url, target, features));
+        const window = this.globalWindow.open(url, target, features);
+
+        // Set name as a unique desktopJS name property instead of overloading window.name
+        window[DefaultContainer.windowNamePropertyKey] = newOptions.name;
+
+        // Add the new window to the global windows object
+        const windows = this.globalWindow[DefaultContainer.windowsPropertyKey];
+        const uuid = window[DefaultContainer.windowUuidPropertyKey] = Guid.newGuid();
+        windows[uuid] = window;
+
+        // Attach unload handler to window to cleanup reference on global windows object
+        window.addEventListener("unload", (event) => delete windows[uuid]);
+
+        // Propagate the global windows object to the new window
+        window[DefaultContainer.windowsPropertyKey] = windows;
+
+        return this.wrapWindow(window);
     }
```

Comment:
There will be work in the future that will revamp the use of the global window variable "desktopJS-windows" used here for maintaining the open windows.  At some point there will be a common container.getAllWindows()